gt/main.c

```diff
@@ -643,8 +611,24 @@ decap_and_fill_eth(struct rte_mbuf *m, struct gt_config *gt_conf,
 			set_ipv4_checksum(&gt_conf->net->front, m, inner_ipv4_hdr);
 		}
 
-		neigh = &instance->neigh;
 		ip_dst = &inner_ipv4_hdr->dst_addr;
+
+		if (!ip4_same_subnet(iface->ip4_addr.s_addr,
+				*(uint32_t *)ip_dst, iface->ip4_mask.s_addr)) {
+			if (inet_ntop(AF_INET, ip_dst,
+					ip, sizeof(ip)) == NULL) {
+				GT_LOG(ERR,
+					"%s: failed to convert a number to an IPv4 address (%s)\n",
+					__func__, strerror(errno));
+				return -1;
+			}
+
+			is_not_neighbor = true;
+			raw_eth = (struct rte_ether_hdr *)pkt_info->l2_hdr;
+			rte_ether_addr_copy(&raw_eth->s_addr, &eth_hdr_tmp.d_addr);
+			rte_ether_addr_copy(&raw_eth->d_addr, &eth_hdr_tmp.s_addr);
```

Comment:
Move this code into the body of `if (is_not_neighbor) {`, so we don't need `eth_hdr_tmp`. That is, just copy the addresses into their final position.